### PR TITLE
Implement first version

### PR DESCRIPTION
**Pull request by deiwin**
From the README:

> # Zipkates: A Kubernetes Metadata Sidecar for Zipkin
>
> Zipkates is a Kubernetes sidecar for Zipkin. It knows about the Pods in your
cluster and uses their metadata to seamlessly add tags to spans.

This implementation features a lightweight (`scratch` + binary) proxy and kubernetes-in-kubernetes integration tests.

Written in Go because of ease of deployment and because it has the best Kuberentes client library.

### CODE REVIEW

**Comment by deiwin** [comment]
It works 🎉

File: README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+LISTEN_PORT       | No       | `9411`               | The port that the proxy will listen for incoming traffic on. Defaults to the default Zipkin port.
+ZIPKIN_PORT       | No       | `9410`               | The port on localhost that the proxy will send traffic to. Has to match the `QUERY_PORT` environment variable of the Zipkin container.
+
+## The name
+
+- Zipkin + Kubernetes
+- Zipkin + k8s
+- Zipkin + k-eight-s
+- Zipkin + kates
+- Zipkates ¯\_(ツ)_/¯
```
Comment (take-five) [inline README.md:97]: image

Right hand is crippled :)

**Comment by take-five** [comment]
As I understand, it's a public project, but is built on our private CI servers. If anyone else outside of Glia wants to contribute, they won't be able to check test failures.
Did you consider building this project on some public CI? It's usually free for open source projects.

File: README.md (hunk 1 above)
Comment (deiwin) [reply README.md]: Heh. https://github.com/salemove/zipkates/compare/9e13037f2db34ffc4cd3a9f45f0103d59b0154c3..0552c9671a8b582f01a3180cce0351b4d598826b

Comment (deiwin) [reply README.md]: Hmm. Now it has short hands 🤔 
Screen Shot 2020-06-01 at 16 04 29

So we need `¯\\\_(ツ)\_/¯` to render ¯\\\_(ツ)\_/¯. This escapes the backslashes and also the underscores used for italics.

https://github.com/salemove/zipkates/compare/0552c9671a8b582f01a3180cce0351b4d598826b..3b43ab302bf595793f14de087d2b8771ada3a535

**Comment by deiwin** [comment]
> If anyone else outside of Glia wants to contribute, they won't be able to check test failures.

In a sense the CI is just `go test ./... && ./test.sh` + publishing. So if tests work locally then it should be good in CI. If it works locally but doesn't on CI, then this can be frustrating for the contributor (I know, I've been in similar situations myself) but shouldn't be an issue most of the time.

> Did you consider building this project on some public CI? It's usually free for open source projects.

Didn't really give it much thought. There are [good examples of getting _kind_ to work on popular CI tools](https://github.com/kind-ci/examples) so it should be quite straight forward to migrate.

So I'm open to changing the CI but as there are other things we still need to figure out re contributions anyway I would do that separately. This way we can already start using this project ourselves until all that is figured out.

**Comment by indrekj** [comment]
> Zipkates is a Kubernetes sidecar for Zipkin. It knows about the Pods in your cluster and uses their metadata to seamlessly add tags to spans.

I'm still somewhat confused what it does. Could you describe why is it useful.

**Comment by deiwin** [comment]
@indrekj, if added as a sidecar to Zipkin, this will allow us to add "owner" (we'll name it differently, though) tags to spans based on the pods the spans were sent from.

We can already do the same for logs and metrics but there was previously no way to do it for traces. So if we set this up then all telemetry (logs, metrics, traces) sent from a pod can be tagged with additional information about service ownership automagically.

Does this clarify things?

**Comment by deiwin** [comment]
An example from the integration test may help understand as well:

1. We [label the pod](https://github.com/salemove/zipkates/blob/0dc000083586f0f04c0fd53ea148ff052844e62b/test.yml#L24) with `custom_label: from_label`.
2. We [configure `LABEL_TAG_MAPPING`](https://github.com/salemove/zipkates/blob/0dc000083586f0f04c0fd53ea148ff052844e62b/test-setup.yml#L76-L77) to `{"custom_label": "custom_tag"}`.
3. We [send two spans through the sidecar](https://github.com/salemove/zipkates/blob/0dc000083586f0f04c0fd53ea148ff052844e62b/test.yml#L64). One with irrelevant tags, one with `custom_tag: "from_data"`.
4. When we [query the spans back from Zipkin](https://github.com/salemove/zipkates/blob/0dc000083586f0f04c0fd53ea148ff052844e62b/test.yml#L79), the first has a tag `custom_tag:  "from_label"` and the second has `custom_tag: "from_data"`.

**Comment by indrekj** [comment]
Ok, that makes sense. There are also globalTags in zipkin clients that could be used but that cannot automatically take all tags (like pod version tags) and we might forget to add global tags to every project.

Just a note, I still think that at some point we should switch to jaeger but that doesn't seem to be happening in the near future.

**Comment by deiwin** [comment]
> There are also globalTags in zipkin clients

About that 😄https://github.com/Financial-Times/tapper/issues/23

Also, Tapper's actually problematic for another reason. I only added support for the v2 API, but Tapper only supports v1. I was planning to propose a PR with v1 support to them.

**Pull request merged**